bag_unpacker/src/main/scala/uk/ac/wellcome/platform/archive/bagunpacker/Unpack.scala

```diff
@@ -0,0 +1,87 @@
+package uk.ac.wellcome.platform.archive.bagunpacker
+
+import java.io.{BufferedInputStream, InputStream, OutputStream}
+
+import grizzled.slf4j.Logging
+import org.apache.commons.compress.archivers.{
+  ArchiveEntry,
+  ArchiveInputStream,
+  ArchiveStreamFactory
+}
+import org.apache.commons.compress.compressors.CompressorStreamFactory
+import org.apache.commons.compress.utils.IOUtils
+
+import scala.util.{Failure, Success, Try}
+
+object Unpack extends Logging {
+
+  def apply(inputStream: InputStream)(
+    out: ArchiveEntry => OutputStream
+  ): Stream[ArchiveEntry] = {
+
+    val readArchive = new ReadArchive(inputStream)
```

Comment:
If I’ve read it correctly, this class can throw if passed an invalid archive - that feels odd for an `apply()` method.